storage_service/locations/models/arkivum.py

```diff
@@ -0,0 +1,172 @@
+from django.db import models
+
+# Third party dependencies, alphabetical
+
+# This project, alphabetical
+from common import utils
+
+# This module, alphabetical
+from . import StorageException
+from location import Location
+from package import Package
+
+LOGGER = logging.getLogger(__name__)
+
+if settings.DEBUG:
+    VERIFY = False
+else:
+    VERIFY = True
+
+
+class Arkivum(models.Model):
+    space = models.OneToOneField('Space', to_field='uuid')
+
+    host = models.CharField(max_length=256,
+        help_text='Hostname of the Arkivum web instance. Eg. arkivum.example.com:8443')
+    # Optionally be able to rsync
+    remote_user = models.CharField(max_length=64, null=True, blank=True,
+        help_text="Optional: Username on the remote machine accessible via passwordless ssh.")
+    remote_name = models.CharField(max_length=256, null=True, blank=True,
+        help_text="Optional: Name or IP of the remote machine.")
+
+    class Meta:
+        verbose_name = "Arkivum"
+        app_label = 'locations'
+
+    ALLOWED_LOCATION_PURPOSE = [
+        Location.AIP_STORAGE,
+    ]
+
+    def browse(self, path):
+        # Support browse so that the Location select works
+        if self.remote_user and self.remote_name:
+            path = os.path.join(path, '')
+            ssh_path = "{}@{}:{}".format(self.remote_user, self.remote_name, utils.coerce_str(path))
+            return self.space._browse_ssh(ssh_path)
+        else:
+            return self.space._browse_local(path)
+
+    def delete_path(self, delete_path):
+        # Can this be done by just deleting the file on disk?
+        # TODO folders
+        url = 'https://' + self.host + '/files/' + delete_path
+        print 'url', url
+        response = requests.delete(url, verify=VERIFY)
+        print 'response', response
+        if response.status_code != 204:
+            raise StorageException('Unable to delete %s', delete_path)
+
+    def move_to_storage_service(self, src_path, dest_path, dest_space):
+        """ Moves src_path to dest_space.staging_path/dest_path. """
+        # Get from watched dir
+        if self.remote_user and self.remote_name:
+            # Rsync from remote
+            src_path = "{user}@{host}:{path}".format(
+                user=self.remote_user,
+                host=self.remote_name,
+                path=src_path)
+        self.space._create_local_directory(dest_path)
+        self.space._move_rsync(src_path, dest_path)
+
+    def move_from_storage_service(self, source_path, destination_path):
+        """ Moves self.staging_path/src_path to dest_path. """
+        # Rsync to Arkivum watched directory
+        if self.remote_user and self.remote_name:
+            rsync_dest = "{user}@{host}:{path}".format(
+                user=self.remote_user,
+                host=self.remote_name,
+                path=destination_path)
+            # Create remote directories
+            command = 'mkdir -p {}'.format(os.path.dirname(destination_path))
```

Comment:
Can we use the remote ssh directory creation from PipelineLocalFS? Maybe break that out into something that's accessible to both?